pytorch_lightning/trainer/trainer.py

```diff
@@ -513,6 +513,8 @@ def train(self):
                                  f' ({self.min_epochs}) or minimum steps ({self.min_steps}) has'
                                  ' not been met. Training will continue...')
 
+            # lower the final step as there is no real change
+            self.global_step -= 1
```

Comment:
why is this here?

Reply:
otherwise, the after 20 steps the values is 20 not 19 which triggers actions you mentioned https://github.com/PyTorchLightning/pytorch-lightning/pull/3807#discussion_r499146397

Reply:
ok I see, an alternative is to not do global_step++ if it is the last batch. I think that would be cleaner but not sure how easy it is. :)